executor/src/witgen/machines/block_machine.rs

```diff
@@ -517,18 +510,6 @@ impl<'a, T: FieldElement> BlockMachine<'a, T> {
         let process_result =
             self.process(mutable_state, &mut sequence_iterator, outer_query.clone())?;
 
-        let process_result = if sequence_iterator.is_cached() && !process_result.is_success() {
```

Comment:
This code tried to solve the same problem.

Disadvantages of the old approach:
- `process_result.is_success()` returns `true` if it was able to complete the outer query. But there might still be cells in the block that could not be solved. They will be set to 0, which might violate some constraints. This causes the issue described in #1559.
- If this block does get executed, the block processor is invoked again from scratch, forgetting any progress it has made already in the first call.